addons/default-costume-editor-color/userscript.js

```diff
@@ -0,0 +1,135 @@
+export default async function ({ addon, global, console, msg }) {
+  // We don"t *need* to wait for the costume editor to be opened, but redux updates take a non-zero
+  // amount of CPU time so let's delay that for as long as possible.
+  await addon.tab.traps.getPaper();
+
+  if (!("colorIndex" in addon.tab.redux.state.scratchPaint.fillMode)) {
+    console.error("Detected new paint editor; this will be supported in future versions.");
+    return;
+  }
+
+  // Removes transparency (which Scratch does not support) and normalizes colors to match how Scratch represents them.
+  const parseColor = (color) => color.substring(0, 7).toUpperCase();
+
+  // Special value Scratch uses as color when objects with different colors are selected
+  // https://github.com/LLK/scratch-paint/blob/6733e20b56f52d139f9885952a57c7da012a542f/src/helper/style-path.js#L10
+  const MIXED = "scratch-paint/style-path/mixed";
+
+  const SCRATCH_DEFAULT_FILL = parseColor("#9966FF");
+  const SCRATCH_DEFAULT_STROKE = parseColor("#000000");
+
+  // color could be a hex color string, MIXED, or null
+  const isValidColor = (color) => typeof color === "string" && color.startsWith("#");
+
+  // returns hex color string, MIXED, or null
+  const getCurrentFillColor = () => addon.tab.redux.state.scratchPaint.color.fillColor.primary;
+  const getCurrentStrokeColor = () => addon.tab.redux.state.scratchPaint.color.strokeColor.primary;
+
+  const getCurrentStrokeWidth = () => addon.tab.redux.state.scratchPaint.color.strokeWidth;
+
+  let defaultFillColor;
+  let defaultStrokeColor;
+  let defaultStrokeWidth;
+  const setDefaultColorsToSettings = () => {
+    defaultFillColor = parseColor(addon.settings.get("fill"));
+    defaultStrokeColor = parseColor(addon.settings.get("stroke"));
+    defaultStrokeWidth = addon.settings.get("strokeSize");
+  };
+  setDefaultColorsToSettings();
+
+  const applyFillColor = () => {
+    if (defaultFillColor !== getCurrentFillColor()) {
+      addon.tab.redux.dispatch({
+        type: "scratch-paint/fill-style/CHANGE_FILL_COLOR",
+        color: defaultFillColor,
+      });
+    }
+  };
+  const applyStrokeColor = () => {
+    if (defaultStrokeColor !== getCurrentStrokeColor()) {
+      addon.tab.redux.dispatch({
+        type: "scratch-paint/stroke-style/CHANGE_STROKE_COLOR",
+        color: defaultStrokeColor,
+      });
+    }
+  };
+  const applyStrokeWidth = () => {
+    if (getCurrentStrokeWidth() !== defaultStrokeWidth) {
+      addon.tab.redux.dispatch({
+        type: "scratch-paint/stroke-width/CHANGE_STROKE_WIDTH",
+        strokeWidth: defaultStrokeWidth,
+      });
+    }
+  };
+  const applyEverything = () => {
+    if (!addon.self.disabled) {
+      applyFillColor();
+      applyStrokeColor();
+      applyStrokeWidth();
+    }
+  };
+  addon.settings.addEventListener("change", () => {
+    setDefaultColorsToSettings();
+    applyEverything();
+  });
```

Comment:
Not sure if this is intentional or not, but changing the default fill color also changes the currently selected fill color in all open editors, even if the default color isn't currently selected.